### PR TITLE
SMB3 upstream, part 6

### PR DESCRIPTION
**Pull request by gwr**
https://www.illumos.org/issues/7529
7529 SMB shares with no permission for root fail after restart
https://www.illumos.org/issues/10981
10981 Can't remove the Domain Admin from the local administrators group.
https://www.illumos.org/issues/10982
10982 Would like an SMB share property to enable/disable quotas
https://www.illumos.org/issues/10983
10983 add mdb walker for smbd share cache
https://www.illumos.org/issues/10984
10984 kernel share list fails to be updated after fs import
https://www.illumos.org/issues/10986
10986 SMB should support "Resource SID Compression"

Next stuff to come is on this branch:
https://github.com/gwr/illumos-gate/commits/smb3d
Handy comparison:
https://github.com/illumos/illumos-gate/compare/master...gwr:smb3d

### CODE REVIEW

File: usr/src/cmd/smbsrv/smbadm/smbadm.c [hunk 1]
```diff
@@ -1408,11 +1408,12 @@ smbadm_group_add_del_member(char *gname, char *mname,
 		 *
 		 * We'e like the real account type if we can get it,
```
Comment (sjorge) [inline usr/src/cmd/smbsrv/smbadm/smbadm.c]: Spelling error? Although not something this PR introduces.

Comment (danmcd) [reply usr/src/cmd/smbsrv/smbadm/smbadm.c]: "We'e" should be "We're"

Comment (jclulow) [reply usr/src/cmd/smbsrv/smbadm/smbadm.c]: Or perhaps _We'd_?

Comment (sjorge) [reply usr/src/cmd/smbsrv/smbadm/smbadm.c]: I assumed `We'e` had to be `We'd` like @jclulow said. Or maybe just `We`

Comment (gwr) [reply usr/src/cmd/smbsrv/smbadm/smbadm.c]: Ah, I see.  Thanks.

**Pull request merged**